apps/homescreen/js/app.js

```diff
@@ -321,6 +324,21 @@ const SETTINGS_VERSION = 0;
   }
 
   App.prototype = {
+    get iconSize() {
+      // If this._iconSize is undefined or 0, let's refresh the value.
+      if (!this._iconSize) {
+        for (var i = 0; i < this.icons.childNodes.length; i++) {
```

Comment:
I totally thought this was already optimised by the platform and this kind of optimisations weren't required anymore. I'll fix it.